docs/tutorials/go-built-in.md

```diff
@@ -56,338 +56,495 @@ We'll start by creating a new Go project.
+information on why the transaction was rejected.
 
-Note that anything with non-zero code will be considered invalid (`-1`, `100`,
-etc.) by Tendermint Core.
+Note that `CheckTx` does not execute the transaction, it only verifies that that the transaction could be executed. We do not know yet if the rest of the network has agreed to accept this transaction into a block.
 
-Valid transactions will eventually be committed given they are not too big and
-have enough gas. To learn more about gas, check out ["the
-specification"](https://github.com/tendermint/tendermint/blob/v0.34.x/spec/abci/apps.md#gas).
 
-For the underlying key-value store we'll use
-[badger](https://github.com/dgraph-io/badger), which is an embeddable,
-persistent and fast key-value (KV) database.
+Finally, make sure to add the bytes package to the `import` stanza at the top of `app.go`:
 
 ```go
-import "github.com/dgraph-io/badger"
+import(
+	"bytes"
 
-type KVStoreApplication struct {
-	db           *badger.DB
-	currentBatch *badger.Txn
-}
-
-func NewKVStoreApplication(db *badger.DB) *KVStoreApplication {
-	return &KVStoreApplication{
-		db: db,
-	}
-}
+	"github.com/dgraph-io/badger/v3"
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+)
 ```
 
-### 1.3.2 BeginBlock -> DeliverTx -> EndBlock -> Commit
 
-When Tendermint Core has decided on the block, it's transfered to the
-application in 3 parts: `BeginBlock`, one `DeliverTx` per transaction and
-`EndBlock` in the end. DeliverTx are being transfered asynchronously, but the
-responses are expected to come in order.
+### 1.3.3 BeginBlock -> DeliverTx -> EndBlock -> Commit
+
+When the CometBFT consensus engine has decided on the block, the block is transferred to the
+application over three ABCI method calls: `BeginBlock`, `DeliverTx`, and `EndBlock`.
+
+- `BeginBlock` is called once to indicate to the application that it is about to
+receive a block.
+- `DeliverTx` is called repeatedly, once for each application transaction that was included in the block.
+- `EndBlock` is called once to indicate to the application that no more transactions
+will be delivered to the application in within this block.
+
+Note that, to implement these calls in our application we're going to make use of Badger's
+transaction mechanism. We will always refer to these as Badger transactions, not to
+confuse them with the transactions included in the blocks delivered by CometBFT,
+the _application transactions_.
+
+First, let's create a new Badger transaction during `BeginBlock`. All application transactions in the
+current block will be executed within this Badger transaction.
+Then, return informing CometBFT that the application is ready to receive application transactions:
 
 ```go
 func (app *KVStoreApplication) BeginBlock(req abcitypes.RequestBeginBlock) abcitypes.ResponseBeginBlock {
-	app.currentBatch = app.db.NewTransaction(true)
+	app.onGoingBlock = app.db.NewTransaction(true)
```

Comment:
It's still called `currentBatch` on v0.34.x

Reply:
Is that a question? I think it is a local variable, right? Defined in line 232.